vertx-pg-client/src/test/java/io/vertx/pgclient/TLSTest.java

```diff
@@ -104,18 +110,30 @@ public void testSslModeDisable(TestContext ctx) {
   @Test
   public void testSslModeAllow(TestContext ctx) {
     Async async = ctx.async();
-    PgConnectOptions options = rule.options()
+    PgConnectOptions options = ruleOptionalSll.options()
       .setSslMode(SslMode.ALLOW);
     PgConnection.connect(vertx, new PgConnectOptions(options)).onComplete(ctx.asyncAssertSuccess(conn -> {
       ctx.assertFalse(conn.isSSL());
       async.complete();
     }));
   }
 
+  @Test
+  public void testSslModeAllowFallback(TestContext ctx) {
+    Async async = ctx.async();
+    PgConnectOptions options = ruleForceSsl.options()
+      .setSslMode(SslMode.ALLOW)
+      .setSslOptions(new ClientSSLOptions().setTrustAll(true));
+    PgConnection.connect(vertx, new PgConnectOptions(options)).onComplete(ctx.asyncAssertSuccess(conn -> {
+      ctx.assertFalse(conn.isSSL());
```

Comment:
that should be `assertTrue` as far as I can tell

Reply:
This should be SSL connection. This is force SSL configuration  (only allows SSL) so the allow mode will first try with no SSL it will fail and it will recover by trying SSL. So if the connection is successful only viable option is SSL true. https://github.com/eclipse-vertx/vertx-sql-client/blob/1bb9d04d42ad050b528925e05e85a269eb328617/vertx-pg-client/src/main/java/io/vertx/pgclient/impl/PgConnectionFactory.java#L97-L98